x-pack/plugins/profiling/public/components/stacked_bar_chart.tsx

```diff
@@ -5,47 +5,109 @@
-
-import { timeFormatter } from '@elastic/charts';
-
+import { TopNSample, TopNSubchart } from '../../common/topn';
+import { chartTheme } from '../utils/chart_styles';
+import { asPercentage } from '../utils/formatters/as_percentage';
 import { TopNContext } from './contexts/topn';
+import { SubChart } from './subchart';
+
+function SubchartTooltip({
+  highlightedSubchart,
+}: TooltipInfo & { highlightedSubchart: TopNSubchart }) {
+  return (
+    <EuiPanel>
+      <SubChart
+        index={highlightedSubchart.Index}
+        color={highlightedSubchart.Color}
+        category={highlightedSubchart.Category}
+        percentage={highlightedSubchart.Percentage}
+        data={highlightedSubchart.Series}
+        height={200}
+        width={400}
+      />
+    </EuiPanel>
+  );
+}
 
 export interface StackedBarChartProps {
-  id: string;
-  name: string;
   height: number;
-  x: string;
-  y: string;
-  category: string;
+  asPercentages: boolean;
+  onBrushEnd: (range: { rangeFrom: string; rangeTo: string }) => void;
+  onSampleClick: (sample: TopNSample) => void;
+  onSampleOut: () => void;
+  highlightedSubchart?: TopNSubchart;
 }
 
 export const StackedBarChart: React.FC<StackedBarChartProps> = ({
-  id,
-  name,
   height,
-  x,
-  y,
-  category,
+  asPercentages,
+  onBrushEnd,
+  onSampleClick,
+  onSampleOut,
+  highlightedSubchart,
 }) => {
   const ctx = useContext(TopNContext);
 
   return (
     <Chart size={{ height }}>
-      <Settings showLegend={false} />
-      <BarSeries
-        id={id}
-        name={name}
-        data={ctx.samples}
-        xAccessor={x}
-        yAccessors={[y]}
-        stackAccessors={[x]}
-        splitSeriesAccessors={[category]}
+      <Settings
+        showLegend={false}
+        tooltip={{
+          customTooltip: highlightedSubchart
+            ? (props) => <SubchartTooltip {...props} highlightedSubchart={highlightedSubchart} />
+            : () => <></>,
+        }}
+        brushAxis={BrushAxis.X}
+        onBrushEnd={(brushEvent) => {
+          const rangeFrom = new Date(brushEvent.x![0]).toISOString();
+          const rangeTo = new Date(brushEvent.x![1]).toISOString();
+
+          onBrushEnd({ rangeFrom, rangeTo });
+        }}
+        theme={chartTheme}
+        onElementClick={(events) => {
+          const [value] = events[0] as XYChartElementEvent;
+          onSampleClick(value.datum as TopNSample);
+        }}
+        onElementOver={() => {
+          onSampleOut();
+        }}
+        onElementOut={() => {
+          onSampleOut();
+        }}
       />
+      {ctx.charts.map((chart) => (
+        <HistogramBarSeries
+          key={chart.Category}
+          id={chart.Category}
+          name={chart.Category}
+          data={chart.Series}
+          color={chart.Color}
+          xAccessor={'Timestamp'}
+          yAccessors={['Count']}
+          stackMode={asPercentages ? StackMode.Percentage : undefined}
```

Comment:
You need to specify the `xScaleType={ScaleType.Time}` and add the proper `timeZone` to make it work correctly, if not, the default scale used for the x axis is a linear scale, that produce not a perfect time axis when used with dates.

Also please consider always to specify the `stackAccessor={[true]}` if your bars are always stacked (even if you are using different stack modes). I know, the `stackAccessor` prop is a bit ugly atm but it will be improved in the future

Reply:
I've added xScaleType and timeZone - but `stackAccessor` is not available as a prop on `StackedBarChart`.

Reply:
Oh true, sorry, the stack mode already does that in this case, np, just forget that part